src/screens/home/home.js

```diff
@@ -1,31 +1,292 @@
+import { Formik } from 'formik';
+import * as Yup from 'yup'
+import { ceil } from 'react-native-reanimated';
+import { _getUser } from '../../utils/asyncUtil';
+import { addContact } from './ducks/home.actions';
+class HomeScreen extends React.Component {
 
-const listItem = () => {
-    return (
-        <ListItem/>
-    )
-}
+    constructor(props) {
+        super(props);
+        this.state = {
+            allUsers: [],
+            modalState: false,
+            search: {},
+            listUsers: []
+        }
+    }
+
+    async componentDidMount() {
+        let user = await _getUser()
+        this.convertUserObjectToArray(this.props.usersProfile, user);
+    }
+
+    convertUserObjectToArray = (usersObj = {}, own) => {
+        let usersArr = [];
+        Object.keys(usersObj).forEach((user = {}) => {
+            if (own !== user) {
+                usersArr.push({ ...usersObj[user], key: user });
+            }
+        });
+        this.setState({ allUsers: usersArr, listUsers: usersArr, mainUser: own });
+    }
+
+    addContact = (singleUser) => {
+        const { allUsers, mainUser } = this.state;
+        console.log('user particular', singleUser);
+        if (singleUser.key) {
+            let updatedArr = singleUser.key;
+            this.props.addContact(mainUser, singleUser);
+            this.setState({ allUsers: allUsers });
+        }
+    }
+
+    listItem = ({ item }) => {
+        const { usersContacts = {} } = this.props;
+        const { mainUser } = this.state;
+        let contacts = usersContacts[mainUser];
+        let isAlreadyinContacts = false;
+        if (contacts && contacts.includes(item.name)) {
+            isAlreadyinContacts = true
+        }
+        if (!item) {
+            return null
+        }
+        return (
+            <ListItem key={item.name} bottomDivider>
+                <Avatar source={item.avatar} />
+                <ListItem.Content>
+                    <ListItem.Title>{item.name + ' (' + item.gender + ')'}</ListItem.Title>
+                    <ListItem.Subtitle>{item.dob}</ListItem.Subtitle>
+                    <View style={styles.subtitleView}>
+                        {/* <Image source={require('../images/rating.png')} style={styles.ratingImage}/> */}
+                        <Text style={styles.ratingText}>{item.location}</Text>
+                    </View>
+                </ListItem.Content>
+                <TouchableOpacity
+                    onPress={() => {
+                        if (!isAlreadyinContacts) {
+                            this.addContact(item);
+                        }
+                    }}
+                >
+                    <ListItem.Chevron
+                        name={isAlreadyinContacts ? 'checkmark-outline' : 'add-circle-outline'}
+                        size={40}
+                    />
+                </TouchableOpacity>
+            </ListItem>
+        )
+    }
+
+    validationSchemaMobile = Yup.object().shape({
```

Comment:
you can create this schema inside a separate file so that we can have a separate view for this and can import it here to use this.